Web/Controller/LandingPages.hs

```diff
@@ -38,6 +38,46 @@ instance Controller LandingPagesController where
                     render EditView { .. }
                 Right landingPage -> do
                     landingPage <- landingPage |> updateRecord
+
+                    -- After we update the Landing page, we can set the order of the paragraphs.
+                    let params = paramListOrNothing @UUID "paragraphId"
+
+                    case catMaybes params of
+                        [] -> do
+                            -- No paragraphs to update.
+                            pure ()
+
+                        uuids -> do
+                            -- We need to update the weight of the paragraphs,
+                            -- So load them.
+                            landingPage <- fetchLandingPageWithParagraphs landingPageId
+
+                            -- Iterate over all paragraphs, and update the weight.
+                            forEach landingPage.paragraphCtas updateParagraph
+
+                            -- @todo: If I uncomment this, it results with a compiler error.
+                            forEach landingPage.paragraphQuotes updateParagraph
+
+                            where
+                                updateParagraph :: forall record.
+                                        ( HasField "id" record (Id record)
+                                        , SetField "weight" record Int
+                                        , CanUpdate record
+                                        , Show (PrimaryKey (GetTableName record))
+                                        , ?modelContext :: ModelContext
+                                        ) => record -> IO ()
+                                updateParagraph paragraph = do
+                                    -- @todo: Why doesn't unpackId paragraph.id work here?
+                                    let uuid = show paragraph.id
```

Comment:
If we try `unpackId` we get an error -- how to solve? Instead I've converted in the line below the Uuids to Text

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/753757a2-8b31-4bc3-a7d7-08ec3c0e0fb0)

Reply:
It misses a constraint `PrimaryKey (GetTableName record) ~ UUID` (IHP also supports non UUID primary keys, so `unpackId` could potentially return something non-UUID, the constraint basically says that the primary key of that record needs to be an UUID).

Here's the full code that works on my clone of this project:

```haskell
                                updateParagraph :: forall record.
                                        ( HasField "id" record (Id record)
                                        , SetField "weight" record Int
                                        , CanUpdate record
                                        , Show (PrimaryKey (GetTableName record))
                                        , ?modelContext :: ModelContext
                                        , PrimaryKey (GetTableName record) ~ UUID
                                        ) => record -> IO ()
                                updateParagraph paragraph = do
                                    let uuid :: UUID = unpackId paragraph.id
                                    let weight = (elemIndex uuid uuids) |> fromMaybe 0

                                    paragraph
                                        |> set #weight weight
                                        |> updateRecord


                                    pure ()
```

Reply:
Great, thank you. I'm new to this `~` - what does it mean in this context?

Reply:
It's an equal constraint, so `a ~ b` means a should be equal to b at the type level